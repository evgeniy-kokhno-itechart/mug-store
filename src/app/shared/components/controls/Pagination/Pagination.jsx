import React, { Component } from 'react';
import PropTypes from 'prop-types';
import _ from 'lodash';
import PaginationButton from '../PaginationButton/PaginationButton';
import './Pagination.scss';

class Pagination extends Component {
  getPages() {
    const { itemsCount, pageSize } = this.props;
    const pagesCount = Math.ceil(itemsCount / pageSize);
    const pages = _.range(1, pagesCount + 1);
    return { pagesCount, pages };
  }

  renderPages(pages) {
    const { currentPage, wrapperClasses } = this.props;

    return (
      <nav className={wrapperClasses}>
        <ul className="pagination-list">
          {pages.map(page => (
            <li key={page} className={currentPage === page ? 'pagination-item--active' : 'pagination-item'}>
              <PaginationButton page={page} handlePageChange={this.props.onPageChange} />
            </li>
          ))}
        </ul>
      </nav>
    );
  }

  render() {
    const { pagesCount, pages } = this.getPages();

    return pagesCount === 1 ? null : this.renderPages(pages);
  }
}

Pagination.propTypes = {
  itemsCount: PropTypes.number.isRequired,
  pageSize: PropTypes.number.isRequired,
  wrapperClasses: PropTypes.string,
  onPageChange: PropTypes.func.isRequired,
  currentPage: PropTypes.number,
};

Pagination.defaultProps = {
  currentPage: 1,
  wrapperClasses: '',
};

export default Pagination;
