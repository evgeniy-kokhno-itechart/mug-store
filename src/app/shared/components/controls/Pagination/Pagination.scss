@import "Variables";

.pagination-list {
    display: flex;
    padding: 0;
    list-style: none;
    margin-bottom: 0;
}

.pagination-item {
    background-image: linear-gradient( to right bottom, rgba($color-gray-light-2, 0.7), rgba($color-gray-light-2, 1));
    border-color: $color-white;
    
    &--active {
        background-image: linear-gradient( to right bottom, rgba($color-primary-dark, 0.7), rgba($color-primary-dark, 1));
        border-color: $color-primary-dark;
        color: $color-white;
    }
}