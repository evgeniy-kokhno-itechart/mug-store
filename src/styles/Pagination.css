#bootstrap-overrides .active > .page-link {
  background-color: gray;
  border-color: gray;
  color: white;
}

#bootstrap-overrides .page-link {
  color: black;
  box-shadow: none;
}
