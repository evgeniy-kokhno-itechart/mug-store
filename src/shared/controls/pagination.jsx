import React, { Component } from 'react';
import PropTypes from 'prop-types';
import _ from 'lodash';
import '../../styles/Pagination.css';

class Pagination extends Component {
  getPages() {
    const { itemsCount, pageSize } = this.props;
    const pagesCount = Math.ceil(itemsCount / pageSize);
    const pages = _.range(1, pagesCount + 1);
    return { pagesCount, pages };
  }

  renderPages(pages) {
    const { currentPage, onPageChange } = this.props;

    return (
      <nav>
        <ul className="pagination">
          {pages.map(p => (
            <li key={p} className={currentPage === p ? 'page-item active' : 'page-item '}>
              <button type="button" className="page-link clickable" onClick={() => onPageChange(p)}>
                {p}
              </button>
            </li>
          ))}
        </ul>
      </nav>
    );
  }

  render() {
    const { pagesCount, pages } = this.getPages();

    return pagesCount === 1 ? null : this.renderPages(pages);
  }
}

Pagination.propTypes = {
  itemsCount: PropTypes.number.isRequired,
  pageSize: PropTypes.number.isRequired,
  onPageChange: PropTypes.func.isRequired,
  currentPage: PropTypes.number,
};

Pagination.defaultProps = {
  currentPage: 1,
};

export default Pagination;
