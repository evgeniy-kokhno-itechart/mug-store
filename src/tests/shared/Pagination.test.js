/* eslint-disable no-undef */
import React from 'react';

import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import Pagination from '../../shared/controls/Pagination';

configure({ adapter: new Adapter() });

describe('<Pagination />', () => {
  let PaginationWrapper;

  beforeEach(() => {
    const onPageChange = jest.fn();
    PaginationWrapper = shallow(<Pagination itemsCount={7} pageSize={5} onPageChange={onPageChange} />);
  });

  it('renders with default props with 1st page checked as active', () => {
    expect(PaginationWrapper.debug()).toMatchSnapshot();
  });

  it('renders and displays 2nd page as active', () => {
    PaginationWrapper.setProps({ currentPage: 2 });
    expect(PaginationWrapper.debug()).toMatchSnapshot();
  });
});
