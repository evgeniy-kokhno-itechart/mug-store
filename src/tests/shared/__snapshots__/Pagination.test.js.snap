// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Pagination /> renders and displays 2nd page as active 1`] = `
"<nav>
  <ul className=\\"pagination\\">
    <li className=\\"page-item \\">
      <button type=\\"button\\" className=\\"page-link clickable\\" onClick={[Function: onClick]}>
        1
      </button>
    </li>
    <li className=\\"page-item active\\">
      <button type=\\"button\\" className=\\"page-link clickable\\" onClick={[Function: onClick]}>
        2
      </button>
    </li>
  </ul>
</nav>"
`;

exports[`<Pagination /> renders with default props with 1st page checked as active 1`] = `
"<nav>
  <ul className=\\"pagination\\">
    <li className=\\"page-item active\\">
      <button type=\\"button\\" className=\\"page-link clickable\\" onClick={[Function: onClick]}>
        1
      </button>
    </li>
    <li className=\\"page-item \\">
      <button type=\\"button\\" className=\\"page-link clickable\\" onClick={[Function: onClick]}>
        2
      </button>
    </li>
  </ul>
</nav>"
`;
